src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,131 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int DEFAULT_SIZE = 0;
+    private static final int START_INDEX = 0;
+    private static final int NEGATIVE_INDEX = -1;
+    private static final int ADD_INDEX_NUMBER = 1;
+    private int capacity;
+    private T[] listT;
+    private int size;
+
+    public ArrayList() {
+        this.capacity = DEFAULT_CAPACITY;
+        this.listT = (T[]) new Object[capacity];
+        size = DEFAULT_SIZE;
+    }
+
     @Override
     public void add(T value) {
-
+        if (capacity > size) {
+            listT[size] = value;
+            size++;
+            return;
+        }
+        increaseCapacity();
+        listT[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (capacity > size) {
```

Comment:
here check index first: 
        `if (index < 0 || index > size) { throw new ArrayListIndexOutOfBoundsException(message)`
        then just paste logic of addToIndex() here. We won't need resizing handling, because new array will be created by System.arraycopy anyway